src/cysignals/implementation.c

```diff
@@ -528,3 +535,361 @@ static void sigdie(int sig, const char* s)
  * signals.pyx. These require some of the above functions, therefore
  * this include must come at the end of this file. */
 #include "macros.h"
+
+#else /* Windows version */
+
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+#include <sys/types.h>
+#include <Python.h>
+#include <stdlib.h>
+#include "struct_signals.h"
+#include <windows.h>
+#include <float.h>
+
+#ifndef __MINGW32__
+#  define inline __inline
+#endif
+/* The cysigs object (there is a unique copy of this, shared by all
+ * Cython modules using cysignals) */
+static cysigs_t cysigs;
+static int win32ctrlc;
+
+static void do_raise_exception(int sig);
+static void sigdie(int sig, const char* s);
+static void print_backtrace(void);
+static void setup_cysignals_handlers(void);
+static void cysigs_interrupt_handler(int sig);
+static void cysigs_signal_handler(int sig);
+
+/* Do whatever is needed to reset the CPU to a sane state after
+ * handling a signals.  In particular on x86 CPUs, we need to clear
+ * the FPU (this is needed after MMX instructions have been used or
+ * if an interrupt occurs during an FPU computation).
+ * Linux and OS X 10.6 do this as part of their signals implementation,
+ * but Solaris doesn't.  Since this code is called only when handling a
+ * signal (which should be very rare), it's better to play safe and
+ * always execute this instead of special-casing based on the operating
+ * system.
+ * See http://trac.sagemath.org/sage_trac/ticket/12873
+ */
+static inline void reset_CPU(void)
+{
+    _fpreset();
```

Comment:
Why are you calling `_fpreset()` here? First of all, it's not entirely clear what it does, but I assume that it resets the complete FPU state, including rounding modes and precision. I don't think that we want this. Does the original cysignals code executing `emms` not work? If you need something extra, we should figure out what you need and do that in a portable way.